components/dash-core-components/src/components/Checklist.react.js

```diff
@@ -33,68 +33,87 @@ export default class Checklist extends Component {
+                                disabled={Boolean(option.disabled)}
+                                style={inputStyle}
+                                type="checkbox"
+                                onChange={() => {
+                                    let newValue;
+                                    if (includes(option.value, value)) {
+                                        newValue = without(
+                                            [option.value],
+                                            value
+                                        );
+                                    } else {
+                                        newValue = append(option.value, value);
+                                    }
+                                    setProps({value: newValue});
+                                }}
+                            />
+                            {option.label}
+                        </label>
+                    );
+                })}
             </div>
         );
     }
 }
 
 Checklist.propTypes = {
-    /**
-     * The ID of this component, used to identify dash components
-     * in callbacks. The ID needs to be unique across all of the
-     * components in an app.
-     */
-    id: PropTypes.string,
-
     /**
      * An array of options
      */
     options: PropTypes.arrayOf(
-        PropTypes.exact({
-            /**
-             * The checkbox's label
-             */
-            label: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
-                .isRequired,
+        PropTypes.oneOfType([
+            PropTypes.exact({
+                /**
+                 * The checkbox's label
+                 */
+                label: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
+                    .isRequired,
+
+                /**
+                 * The value of the checkbox. This value
+                 * corresponds to the items specified in the
+                 * `value` property.
+                 */
+                value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
+                    .isRequired,
+
+                /**
+                 * If true, this checkbox is disabled and can't be clicked on.
+                 */
+                disabled: PropTypes.bool,
+            }),
 
             /**
-             * The value of the checkbox. This value
-             * corresponds to the items specified in the
-             * `value` property.
+             * We now accept the single string `value` as an option value,
+             * which equals to
+             * { label: `value`, value: `value`, disabled: false }
              */
-            value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
-                .isRequired,
-
-            /**
-             * If true, this checkbox is disabled and can't be clicked on.
-             */
-            disabled: PropTypes.bool,
-        })
+            PropTypes.string,
```

Comment:
Is it necessary to limit this to a string only?  Or could also include numbers?:
`PropTypes.oneOfType([PropTypes.string, PropTypes.number])`

Reply:
Great point @AnnMarieW ! `bool` while we're at it too

Reply:
@AnnMarieW Added `PropTypes.number` and use the number as a label of the option. 74c25af5
@chriddyp But question on `bool`: Is it okay to have empty `label` but with only `value` in an option?

Reply:
For bool, I think we'd just convert the bools into strings, so `[{'label': 'True', 'value': True}, {'label': 'False', 'value': False}]`